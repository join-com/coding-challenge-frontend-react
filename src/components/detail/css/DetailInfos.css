.DetailInfos:nth-child(odd) {
  background-color: var(--list-element-odd-background-color);
  margin: var(--section-margin) 0;
}

.DetailInfos__container {
  display: flex;
}

.DetailInfos__infos {
  padding: var(--section-padding) 0;
  display: flex;
  flex-wrap: wrap;
}

.DetailInfos__property {
  display: flex;
  flex-wrap: wrap;
  padding: var(--element-padding) 0;
  width: 100%;
  flex-basis: 100%;
  margin-right: calc(var(--section-margin) / 2);
}

.DetailInfos__property__label {
  margin-right: calc(var(--section-margin) / 2);
}

.DetailInfos__property--title {
  font-weight: 500;
  font-size: 1.4em;
}

.DetailInfos__property--occurred_at,
.DetailInfos__property--address {
  width: auto;
  flex-basis: auto;
}

.DetailInfos__property--address {
  padding-left: 0;
}

.DetailInfos__property--description .DetailInfos__property__label {
  text-transform: uppercase;
  font-size: 1.4em;
  margin: var(--section-margin) 0;
}

.DetailInfos__property--description {
  width: 100%;
  flex-basis: 100%;
}
