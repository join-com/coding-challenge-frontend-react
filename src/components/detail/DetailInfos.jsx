import React from 'react';

import { formatKeyValueByType } from '../../global/utils/FormatHelpers';

import I18n from '../../utils/I18n';

import './css/DetailInfos.css';

const keys = [
  {
    name: 'title',
    label: '',
    type: 'string',
  },
  {
    name: 'occurred_at',
    label: I18n.t('en.data.attribute.occurredAt.label'),
    type: 'date',
    format: 'ddd MMM DD YYYY',
  },
  {
    name: 'address',
    label: I18n.t('en.data.attribute.address.label'),
    type: 'string',
  },
  {
    name: 'description',
    label: I18n.t('en.data.attribute.description.label'),
    type: 'string',
  },
];

const DetailInfos = ({ item }) => {
  if (item) {
    return (
      <div className="DetailInfos">
        <div className="DetailInfos__container">
          <div className="DetailInfos__infos">
            {keys.map(key => (
              <div
                className={`DetailInfos__property DetailInfos__property--${key.name}`}
                key={key.name}
              >
                {key.label
                  && <div className="DetailInfos__property__label">{key.label}</div>
                }
                <div className="DetailInfos__property__value">
                  {formatKeyValueByType(key, item)}
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }

  return null;
};

export default DetailInfos;
