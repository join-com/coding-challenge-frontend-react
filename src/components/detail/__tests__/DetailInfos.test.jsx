import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import DetailInfos from '../DetailInfos';

const { incidents } = staticData;

const props = {
  item: incidents[0],
};

describe('DetailInfos', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<DetailInfos {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component if no item', () => {
    const tree = shallow(<DetailInfos {...props} item={null} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
