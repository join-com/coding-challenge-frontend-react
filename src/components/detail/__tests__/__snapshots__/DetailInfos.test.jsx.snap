// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailInfos should not render the component if no item 1`] = `""`;

exports[`DetailInfos should render the component correctly 1`] = `
<div
  className="DetailInfos"
>
  <div
    className="DetailInfos__container"
  >
    <div
      className="DetailInfos__infos"
    >
      <div
        className="DetailInfos__property DetailInfos__property--title"
        key="title"
      >
        <div
          className="DetailInfos__property__value"
        >
          Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)
        </div>
      </div>
      <div
        className="DetailInfos__property DetailInfos__property--occurred_at"
        key="occurred_at"
      >
        <div
          className="DetailInfos__property__label"
        >
          Stolen
        </div>
        <div
          className="DetailInfos__property__value"
        >
          Tue Nov 27 2018
        </div>
      </div>
      <div
        className="DetailInfos__property DetailInfos__property--address"
        key="address"
      >
        <div
          className="DetailInfos__property__label"
        >
          CET at
        </div>
        <div
          className="DetailInfos__property__value"
        >
          Berlin, 12051, DE
        </div>
      </div>
      <div
        className="DetailInfos__property DetailInfos__property--location"
        key="location"
      >
        <div
          className="DetailInfos__property__value"
        >
          <MapWrapper
            name="location"
            zoom={15}
          />
        </div>
      </div>
      <div
        className="DetailInfos__property DetailInfos__property--description"
        key="description"
      >
        <div
          className="DetailInfos__property__label"
        >
          Description of incident
        </div>
        <div
          className="DetailInfos__property__value"
        >
          Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. 
        </div>
      </div>
    </div>
  </div>
</div>
`;
